tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -436,6 +382,67 @@ Status StorageManager::array_open_for_writes(
   return Status::Ok();
 }
 
+Status StorageManager::array_reopen(
```

Comment:
Let's call this `array_load_fragments` to be consistent with my suggestion above in `Array`.